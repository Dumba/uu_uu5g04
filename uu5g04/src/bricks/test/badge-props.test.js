/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from "react";
import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.ColorSchemaMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin"
  ],
  props: {

  },
  requiredProps: {},
  opt: {
  }
};


describe(`UU5.Bricks.Badge`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Badge, CONFIG);
});

describe(`UU5.Bricks.Badge badge is nested in another Component`, () => {

  it(`UU5.Bricks.Badge with Icon`, () => {
    const tree = shallow(
      <UU5.Bricks.Container id={"uuID"}>
      <UU5.Bricks.TouchIcon id={"uuID2"} icon="mdi-account-multiple" bgStyle="transparent" className="icon" />
      <UU5.Bricks.Badge id={"uuID3"} content="3" className="badge1" colorSchema="red" />
      </UU5.Bricks.Container>
    );
    expect(tree).toMatchSnapshot();
  });

  it(`UU5.Bricks.Badge redner into Link`, () => {
    const tree = shallow(
      <UU5.Bricks.Link id={"uuID"} href="#">Updates
        <UU5.Bricks.Badge id={"uuID2"} content="4" />
      </UU5.Bricks.Link>
    );
    expect(tree).toMatchSnapshot();
  });

  it(`UU5.Bricks.Badge render into Button`, () => {
    const tree = shallow(
      <UU5.Bricks.Button id={"uuID"} size="l" colorSchema="primary">
        Fresh
        <UU5.Bricks.Badge id={"uuID2"} content="152" colorSchema="purple" />
      </UU5.Bricks.Button>
    );
    expect(tree).toMatchSnapshot();
  });

});


