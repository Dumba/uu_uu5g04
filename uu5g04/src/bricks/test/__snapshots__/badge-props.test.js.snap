// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Badge badge is nested in another Component UU5.Bricks.Badge redner into Link 1`] = `
<a
  className="uu5-bricks-link"
  download={false}
  href="http://example.com/test_url_for_tools-ifc.test.js?param=value#"
  id="uuID"
  style={Object {}}
  target="_self"
  title={null}
>
  <UU5.Common.TextCorrectorContextConsumer
    key="0"
    text="Updates"
  />
  <Badge
    content="4"
    id="uuID2"
    key="uuID2"
  />
</a>
`;

exports[`UU5.Bricks.Badge badge is nested in another Component UU5.Bricks.Badge render into Button 1`] = `
<button
  className="color-schema-blue-rich color-schema-primary uu5-bricks-button uu5-bricks-button-l uu5-bricks-button-filled"
  id="uuID"
  style={Object {}}
  title={null}
  type="button"
>
  <span
    className="uu5-bricks-button-text"
    key="0"
  >
    <UU5.Common.TextCorrectorContextConsumer
      key="0"
      text="Fresh"
    />
  </span>
  <Badge
    colorSchema="purple"
    content="152"
    id="uuID2"
    key="uuID2"
  />
</button>
`;

exports[`UU5.Bricks.Badge badge is nested in another Component UU5.Bricks.Badge with Icon 1`] = `
<div
  className="uu5-bricks-container uu5-bricks-container-spacing"
  id="uuID"
  style={Object {}}
  title={null}
>
  <TouchIcon
    bgStyle="transparent"
    className="icon"
    icon="mdi-account-multiple"
    id="uuID2"
    key="uuID2"
  />
  <Badge
    className="badge1"
    colorSchema="red"
    content="3"
    id="uuID3"
    key="uuID3"
  />
</div>
`;

exports[`UU5.Bricks.Badge checkGrammar=false 1`] = `""`;

exports[`UU5.Bricks.Badge checkGrammar=true 1`] = `""`;

exports[`UU5.Bricks.Badge checkHighlight=false 1`] = `""`;

exports[`UU5.Bricks.Badge checkHighlight=true 1`] = `""`;

exports[`UU5.Bricks.Badge checkSpaces=false 1`] = `""`;

exports[`UU5.Bricks.Badge checkSpaces=true 1`] = `""`;

exports[`UU5.Bricks.Badge className="my-classname" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="brown" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="brown-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="cyan" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="cyan-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="default" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="green" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="green-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="grey" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="grey-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="orange" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="orange-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="pink" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="pink-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="purple" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="purple-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="red" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="red-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="yellow" 1`] = `""`;

exports[`UU5.Bricks.Badge colorSchema="yellow-rich" 1`] = `""`;

exports[`UU5.Bricks.Badge content="This is Content from ContentMixin" 1`] = `
<span
  className="uu5-bricks-badge"
  id="uuID"
  style={Object {}}
  title={null}
>
  <UU5.Common.TextCorrectorContextConsumer
    key="0"
    text="This is Content from ContentMixin"
  />
</span>
`;

exports[`UU5.Bricks.Badge content=12345 1`] = `
<span
  className="uu5-bricks-badge"
  id="uuID"
  style={Object {}}
  title={null}
>
  12345
</span>
`;

exports[`UU5.Bricks.Badge controlled=false 1`] = `""`;

exports[`UU5.Bricks.Badge controlled=true 1`] = `""`;

exports[`UU5.Bricks.Badge default props 1`] = `""`;

exports[`UU5.Bricks.Badge disabled=false 1`] = `""`;

exports[`UU5.Bricks.Badge disabled=true 1`] = `""`;

exports[`UU5.Bricks.Badge dynamic=false 1`] = `""`;

exports[`UU5.Bricks.Badge dynamic=true 1`] = `""`;

exports[`UU5.Bricks.Badge hidden=false 1`] = `""`;

exports[`UU5.Bricks.Badge hidden=true 1`] = `""`;

exports[`UU5.Bricks.Badge id="myId" 1`] = `""`;

exports[`UU5.Bricks.Badge ignoreHTML=false 1`] = `""`;

exports[`UU5.Bricks.Badge ignoreHTML=true 1`] = `""`;

exports[`UU5.Bricks.Badge mainAttrs=undefined 1`] = `""`;

exports[`UU5.Bricks.Badge name="myName" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="bigBox" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="bigBoxCollection" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="box" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="boxCollection" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="container" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="inline" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="page" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="smallBox" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="smallBoxCollection" 1`] = `""`;

exports[`UU5.Bricks.Badge nestingLevel="spa" 1`] = `""`;

exports[`UU5.Bricks.Badge noIndex=false 1`] = `""`;

exports[`UU5.Bricks.Badge noIndex=true 1`] = `""`;

exports[`UU5.Bricks.Badge pureRender=false 1`] = `""`;

exports[`UU5.Bricks.Badge pureRender=true 1`] = `""`;

exports[`UU5.Bricks.Badge selected=false 1`] = `""`;

exports[`UU5.Bricks.Badge selected=true 1`] = `""`;

exports[`UU5.Bricks.Badge style="padding:8px; borderRadius: 2px" 1`] = `""`;

exports[`UU5.Bricks.Badge style={"backgroundColor":"red","color":"blue"} 1`] = `""`;

exports[`UU5.Bricks.Badge textCorrector=false 1`] = `""`;

exports[`UU5.Bricks.Badge textCorrector=true 1`] = `""`;

exports[`UU5.Bricks.Badge tooltip="My Tooltip" 1`] = `""`;
